### DISCUSSION TITLE
Retrieve messages from SQLite

### DISCUSSION THREAD

**Discussion by WouterHouwen** [nanomq/nanomq#1622] [Q&A] [answered]
I've setup a docker container that has some data persistence rules to SQLite and message are persisted to the SQLite file.

Is there something inside NanoMQ that can given me access to the messages that are persisted to the SQLite file?

I know you can use sqlite3 to open the database, but I want to create an application that can work with the persisted data that doesn't run inside the NanoMQ container

**Comment by JaylinYu** [comment, marked as answer]
Directly access the database even modify the table gonna cause unstable issue of nanomq. the sqlite table (msg caching) is a feature designed for internal use only.

As for rule-engine/data-persistence, this is totally open, doesn't require anything inside of nanomq, just read the sqlite database file.
